tests/framework/matcher/unstructured_bool.go

```diff
@@ -0,0 +1,50 @@
+package matcher
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/onsi/gomega/types"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"kubevirt.io/kubevirt/tests/framework/matcher/helper"
+)
+
+func MatchUnstructuredBool(fields ...string) types.GomegaMatcher {
```

Comment:
What do you think about creating a generic matcher that would use unstructured to access a field, and then run another matcher on that field. Similarly to the `WithTransformMatcher` from gomega.
It could have signature like:
```go
func Field(patch []string, matcher types.GomegaMatcher) types.GomegaMatcher { ... }
```
And be used like:
```go
Expect(vmi).To(Field([]string{"status", "phase"}, Equal("Running")))
```

Reply:
I like this idea! Will post an updated PR soon.